### PR TITLE
AK: Make single pivot quick_sort guarantee a max stack depth of log(n)

### PR DESCRIPTION
**Pull request by Maato**
- The change to quick_sort requires SimpleIterator to support
assignment.

### CODE REVIEW

File: AK/QuickSort.h [hunk 1]
```diff
@@ -88,27 +88,36 @@ void dual_pivot_quick_sort(Collection& col, int start, int end, LessThan less_th
...
-
-    if (pivot_point)
-        swap(*(start + pivot_point), *start);
+    for (;;) {
+        int size = end - start;
+        if (size <= 1)
+            return;
+
+        int pivot_point = size / 2;
+        auto pivot = *(start + pivot_point);
```
Comment (alimpfard) [inline AK/QuickSort.h:97]: Pls no copying the pivot?
I thought I made a PR with that change a while back 🤔

Comment (Maato) [reply AK/QuickSort.h:97]: That was already there, maybe you changed that in the dual_pivot_quick_sort?

Comment (alimpfard) [reply AK/QuickSort.h:97]: Ooooh, you're right!
I missed the single pivot one, it seems ^^'

Well...please make that a forwarding reference while you're at it :)

Comment (Maato) [reply AK/QuickSort.h]: Changed it, also included it in the test case.

**Pull request merged**